codice/ddf
# DDF-2438 Fixed registry entry duplication issue in IdentificationPlugin

#### What does this PR do?
Fixes a duplication issue where loopback registry configurations cause local nodes to be duplicated.
There had been duplication logic in the IdentificationPlugin before but was removed when we
added the internal registry entries. Added the logic back in a separate class and only apply it to registry entries that aren't internal.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@mcalcote @vinamartin 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@stustison 
@figliold
#### How should this be tested?
Install ddf and ddf-registry. Create a loopback remote registry. After a minute or so go to solr and verify there is only one registry metacard entry (metacard-tags=registry)
#### Any background context you want to provide?
#### What are the relevant tickets?
DDF-2438
#### Screenshots (if appropriate)
#### Checklist:
